examples/chip-tool/commands/common/Command.cpp

```diff
@@ -98,14 +122,40 @@ bool Command::InitArguments(int argc, char ** argv)
             {
                 found = true;
 
-                VerifyOrExit((size_t) argc > (i + 1),
-                             ChipLogError(chipTool, "InitArgs: Optional argument %s missing value.", argv[i]));
-                if (!InitArgument(j, argv[i + 1]))
+                if (mArgs[j].isNullable())
                 {
-                    ExitNow();
+                    if ((size_t) argc <= (i + 1))
+                    {
+                        // This is the last argument, so set it to null.
+                        VerifyOrDo(InitArgument(j, kOptionalArgumentNullableDefault), ExitNow());
+                        continue;
+                    }
+
+                    if (strncmp(argv[i + 1], kOptionalArgumentPrefix, kOptionalArgumentPrefixLength) == 0)
+                    {
+                        // The argument is followed by an other optional argument, so set it to null.
```

Comment:
This breaks when an optional argument is nullable and of string type and the desired string value happens to start with "--", no?

Reply:
I guess you could do `--foo "str:--value"`?